### DISCUSSION TITLE
Manually removing many files from statistics

### DISCUSSION THREAD

**Discussion by Xevion** [github-linguist/linguist#5902] [Classification] [answered]
I'm trying to remove a bunch of HTML files from my repository's language statistics. Currently, it's marked as HTML but it's all data files that are not useful to describe my Vue-based application's primary language (Vue).

[Repository Link](https://github.com/Xevion/the-office/)
[Current Live .gitattributes](https://github.com/Xevion/the-office/blob/master/.gitattributes)

I tried all of the following configurations with no luck. None of them removed the files from statistics.

```gitattributes
server/normalization/html/* linguist-documentation
server/normalization/html/* linguist-vendored
server/normalization/html/* linguist-vendored=false
server/normalization/html/* -linguist-vendored
server/normalization/html/*.html linguist-vendored
server/normalization/html/** linguist-vendored
```

Here is what `github-linguist`'s results look like locally for me.

```python
$ github-linguist -b
99.64%  25537836   HTML
0.17%   42353      Python
0.12%   31547      Vue
0.05%   12808      JavaScript
0.02%   5977       SCSS
0.00%   336        TypeScript

JavaScript:
.eslintrc.js
babel.config.js
functions/.eslintrc.js
src/main.js
src/mutation_types.js
src/router.js
src/store.js
vue.config.js

TypeScript:
functions/src/index.ts

HTML:
public/index.html
server/normalization/html/1-01.html
server/normalization/html/1-02.html
server/normalization/html/1-03.html
server/normalization/html/1-04.html
server/normalization/html/1-05.html
server/normalization/html/1-06.html
server/normalization/html/2-01.html
[shortened manually]

Python:
server/api.py
server/normalization/helpers.py
server/normalization/main.py

Vue:
src/App.vue
src/components/Character.vue
src/components/CharacterBadges.vue
src/components/Characters.vue
src/components/Episode.vue
src/components/Home.vue
src/components/QuoteList.vue
src/components/SearchResult.vue
src/components/SearchResults.vue
src/components/Season.vue
src/components/SeasonList.vue
src/components/SeasonListItem.vue
src/components/Skeleton.vue

SCSS:
src/scss/_variables.scss
src/scss/main.scss
```

**Comment by Xevion** [comment]
Okay, I just checked now and suddenly the repositories language statistics are back to where I want them.
Still though, I do not understand why my local `github-linguistics` wasn't representative of the configuration I was running.
Does `github-linguistics` not take `.gitattributes` into effect when ran locally? Do I have to somehow explicitly configure it to use it, or does it use a different file? This whole override thing is so niche and I really am having a hard time wrapping my head around what's going on lol.

**Comment by lildude** [comment, marked as answer]
> Okay, I just checked now and suddenly the repositories language statistics are back to where I want them.

Simply changing the `.gitattributes` is not always enough to trigger a re-analysis. You may need to push another change to trigger the analysis. The analysis also takes place in the background ([documented](https://github.com/github/linguist/blob/master/docs/how-linguist-works.md#how-linguist-works-on-githubcom)) so may take a bit of time during busy periods. It can also take time for all caches to be updated. Both of these are out of Linguist's control.

> Does `github-linguistics` not take `.gitattributes` into effect when ran locally? Do I have to somehow explicitly configure it to use it, or does it use a different file?

It does work locally but you need to commit it to your repo for it to take effect. This is [documented](https://github.com/github/linguist/blob/master/docs/overrides.md#using-gitattributes).

**Reply by Xevion** [reply]
> It does work locally but you need to commit it to your repo for it to take effect. This is [documented](https://github.com/github/linguist/blob/master/docs/overrides.md#using-gitattributes).

I see. That's what had me all confused. I did not anticipate nor read that the `.gitattributes` would need to be committed for the **local** command to work properly.
